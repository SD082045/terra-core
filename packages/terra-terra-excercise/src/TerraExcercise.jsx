import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import classNamesBind from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import styles from './TerraExcercise.module.scss';

const cx = classNamesBind.bind(styles);

const propTypes = {
  /**
   * Content to be displayed as the name
   */
  name: PropTypes.string,
};

const defaultProps = {
  name: 'Red Button',
};

const TerraExcercise = ({ name, ...customProps }) => {
  const theme = React.useContext(ThemeContext);
  const TerraExcerciseClassNames = classNames(
    cx([
    'terra-excercise',
    theme.className,
  ]),
    customProps.className,
  );

  return (<button {...customProps} className={TerraExcerciseClassNames}>{name}</button>);
};

TerraExcercise.propTypes = propTypes;
TerraExcercise.defaultProps = defaultProps;

export default TerraExcercise;
