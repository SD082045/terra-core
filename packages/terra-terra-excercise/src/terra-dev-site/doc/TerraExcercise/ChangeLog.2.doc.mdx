import { Badge } from 'terra-terra-excercise/package.json?dev-site-package';
import ChangeLog from 'terra-terra-excercise/CHANGELOG.md';

<Badge />

<ChangeLog />
