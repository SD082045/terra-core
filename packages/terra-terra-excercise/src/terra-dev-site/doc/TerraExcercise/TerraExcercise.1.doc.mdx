import { Badge } from 'terra-terra-excercise/package.json?dev-site-package';

import DefaultTerraExcercise from '../example/DefaultTerraExcercise?dev-site-example';

import PropsTable from 'terra-terra-excercise/src/TerraExcercise?dev-site-props-table';

<Badge />

# Terra Terra Excercise

{insert description}

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-terra-excercise`

## Usage

```jsx
import React from 'react';
import TerraExcercise from 'terra-terra-excercise';

<TerraExcercise {props...} />
```

## Component Features

* [Cross-Browser Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#cross-browser-support)
* [LTR/RTL Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#ltr--rtl-support)

## Examples
<DefaultTerraExcercise />

## Divider Props
<PropsTable />
