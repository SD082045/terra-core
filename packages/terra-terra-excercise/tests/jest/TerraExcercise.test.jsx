import React from 'react';
import TerraExcercise from '../../src/TerraExcercise';

describe('TerraExcercise', () => {
  const defaultRender = <TerraExcercise />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.find('.terra-excercise').text()).toEqual('default');
  });

  // Structure Tests
  it('should have the class terra-excercise', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('terra-excercise');
  });
});
