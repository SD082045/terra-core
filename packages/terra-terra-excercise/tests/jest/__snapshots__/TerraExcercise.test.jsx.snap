// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerraExcercise should render a default component 1`] = `
<button
  className="terra-excercise"
>
  Red Button
</button>
`;
